### PR TITLE
DOC: Document Forward and SourceSpaces

### LINKED ISSUE #10901: List of dict values for Forward object
Would it be possible to document the various values of the Forward object? For example, if `fwd = mne.Forward()` it would be nice to have a succinct list showing the differences between `fwd['src']` and `fwd['src'][0]` and `fwd['src'][0]['vertno']` without having to do a `.__dict__()` search for each level of information.

### PR DESCRIPTION
**Pull request by larsoner**
Closes #10901

@jadrew43 can you look and see if these are readable enough?

### CODE REVIEW

**Comment by larsoner** [comment]
Rendered versions will be easier to read @jadrew43 :

- [Forward](https://output.circle-artifacts.com/output/job/9fde3fe0-b3ff-4f71-8da7-61d66025b431/artifacts/0/dev/generated/mne.Forward.html)
- [SourceSpaces](https://output.circle-artifacts.com/output/job/9fde3fe0-b3ff-4f71-8da7-61d66025b431/artifacts/0/dev/generated/mne.SourceSpaces.html)

**Comment by jadrew43** [comment]
That's helpful thanks.

**Comment by larsoner** [comment]
@drammock feel free to merge if you're happy or push wording changes directly

File: mne/forward/forward.py [hunk 1]
```diff
@@ -54,12 +54,77 @@
...
+            The forward solution, with entries:
+
+            ``'data'`` : ndarray, shape (n_channels, nsource * n_ori)
+                The forward solution data. The shape will be
+                ``(n_channels, nsource)`` for a fixed-orientation forward and
+                ``(n_channels, nsource * 3)`` for a free-orientation forward.
+            ``'row_names'`` : list of str
+                The channel names.
+        mri_head_t : instance of Transform
+            The mri<->head transformation that was used.
```
Comment (drammock) [inline mne/forward/forward.py:101]:
```suggestion
            The mri ↔ head transformation that was used.
```

**Comment by larsoner** [comment]
> Something is wrong with how Attributes section is getting rendered.

That is not due to this PR, that's how numpydoc/pydata-sphinx-theme etc. do it. Same problem exists on https://mne.tools/dev/generated/mne.Forward.html

File: mne/source_space.py [hunk 2]
```diff
@@ -68,9 +74,127 @@ class SourceSpaces(list):
...
+    Notes
+    -----
+    Each element in SourceSpaces (e.g., ``src[0]``) is a dictionary. For
+    example, a surface source space will have ``len(src) == 2``, one entry for
+    each hemisphere. A volume source space will have ``len(src) == 1`` if it
+    uses a single monolithic grid, or ``len(src) == len(volume_label)`` when
+    created with a list-of-atlas-labels. A mixed source space consists of both
+    surface and volumetric source spaces in a single SourceSpaces object.
+
+    Each element in the SourceSpaces list (e.g., ``src[0]``) is a dictionary.
```
Comment (drammock) [inline mne/source_space.py:99]:
```suggestion
```

The first sentence of prev. paragraph already said this.

File: mne/source_space.py [hunk 3]
```diff
@@ -68,9 +74,127 @@ class SourceSpaces(list):
...
+            For each vertex on the dense surface, this gives the vertex index
+            on the subsampled surface that it's closest to.
+        nearest_dist : ndarray, shape (np,)
+            The distances corresponding to ``nearest``.
+
+    **Volume source spaces**
+
+    Volume source spaces created using :func:`mne.setup_volume_source_space`
+    can have the following additional entries (which will be missing, or
+    have values of None or 0 for surface source spaces):
```
Comment (drammock) [inline mne/source_space.py:170]:
```suggestion
    have values of ``None`` or ``0`` for surface source spaces):
```

File: mne/source_space.py [hunk 4]
```diff
@@ -68,9 +74,127 @@ class SourceSpaces(list):
...
+            The vertex numbers of the dense surface or complete volume that are
+            used (i.e., ``np.where(src[0]['inuse'])[0]``).
+        subject_his_id : str
+            The FreeSurfer subject name.
+
+    **Surface source spaces**
+
+    Surface source spaces created using :func:`mne.setup_source_space` can have
+    the following additional entries (which will be missing, or have values of
+    None or 0 for volumetric source spaces):
```
Comment (drammock) [inline mne/source_space.py:138]:
```suggestion
    ``None`` or ``0`` for volumetric source spaces):
```

**Review by drammock — APPROVED** [review]

**Comment by drammock** [comment]
waiting for last CI to pass, then will commit the cosmetic suggestions & merge.

**Comment by drammock** [comment]
thx @larsoner

**Pull request merged**